File: Day45/GCDofString.java
import java.util.*;

public class GCDofString {
    public String gcdOfStrings(String str1, String str2) {
        int a = str1.length(), b = str2.length();
        while (b != 0) {
            int tmp = b;
            b = a % b;
            a = tmp;
        }
        if (!(str1 + str2).equals(str2 + str1)) return "";
        return str1.substring(0, a);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str1 = sc.nextLine();
        String str2 = sc.nextLine();
        System.out.println(new GCDofString().gcdOfStrings(str1, str2));
    }
}

File: Day45/GCDofString.py
class Solution(object):
    def gcdOfStrings(self, str1, str2):
        a, b = len(str1), len(str2)
        while b:
            a, b = b, a % b
        if str1 + str2 != str2 + str1:
            return ""
        return str1[:a]

if __name__ == "__main__":
    str1 = input().strip()
    str2 = input().strip()
    sol = Solution()
    print(sol.gcdOfStrings(str1, str2))
